cheetah/accelerator/transverse_deflecting_cavity.py

```diff
@@ -170,16 +170,24 @@ def _track_bmadx(self, incoming: ParticleBeam) -> ParticleBeam:
             )
         )
 
-        px = px + voltage * torch.sin(phase)
+        # TODO: Assigning px to px is really bad practice and should be separated into
+        # two separate variables
+        px = px + voltage.unsqueeze(-1) * torch.sin(phase)
```

Comment:
Why do we need `.unsqueeze(-1)` here and (I think also the quadrupole bmad tracking)? I'm probably just confused.

W.r.t. the variable naming, it can probably be `px_new` or `px_out`

Reply:
Because `px` has a shape`(vector, num_particles)` and `voltage` has `(vector, )` ... for PyTorch broadcasting the vector dimension of `voltage` basically has to be pushed to the left ... and that's what the `unsqueeze` does.

All the Bmad-X code needs refactoring, including that variable name ... I've had to keep myself from doing that in this PR.

Reply:
I see! Then we should merge this first and do the refactoring later.